mage_ai/frontend/pages/datasets/[slug].tsx

```diff
@@ -132,6 +152,40 @@ function Data() {
   // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [statistics]);
 
+  // updates suggestions and filters any removed or applied actions
+  useEffect(() => {
+    const filteredSuggestions = [...suggestionsMemo];
+    removedSuggestions.forEach(i => filteredSuggestions.splice(i, 1));
+    actions.forEach(({ i }) => filteredSuggestions.splice(i, 1));
+    setSuggestions(filteredSuggestions);
+  }, [
+    actions,
+    suggestionsMemo,
+    removedSuggestions,
+  ]);
+
+  const addAction = i => {
+    setActions(actions.concat({ i, suggestions: suggestions[i] }));
+  };
+
+  const removeAction = i => {
+    setActions(actions.filter((x, idx) => i !== idx));
+  }
+
+  const removeSuggestion = i => {
```

Comment:
I believe we keep track of skipped/removed suggestions with the suggestion's `status`, which needs to be persisted, rather than just using the React state.

Reply:
Yes, this was my biggest concern, if the functionality of suggestions/actions lined up with how the data cleaning tool should actually work. I think we should check in with Xiaoyou/David ASAP if everything works the way the backend intended